librz/bin/format/luac/luac_bin.c

```diff
@@ -16,15 +17,17 @@ void luac_add_section(RzList *section_list, char *name, ut64 offset, ut32 size,
 	bin_sec->bits = is_func ? sizeof(LUA_INSTRUCTION) * 8 : 8;
 	// bin_sec->has_strings = !is_func;
 	bin_sec->has_strings = false;
-	bin_sec->arch = rz_str_new("luac");
+	bin_sec->arch = "luac";
 
 	if (is_func) {
 		bin_sec->perm = RZ_PERM_R | RZ_PERM_X;
 	} else {
 		bin_sec->perm = RZ_PERM_R;
 	}
 
-	rz_list_append(section_list, bin_sec);
+	if (!rz_list_append(section_list, bin_sec)) {
+		rz_bin_section_free(bin_sec);
```

Comment:
abort(); ?

Reply:
no need. tbh, the code needs a lot of love.